src/getindex.jl

```diff
@@ -0,0 +1,113 @@
+    nothing
+end
+function Base.checkbounds(A::LinearMap, i)
+    Base.checkindex(Bool, Base.OneTo(length(A)), i) || throw(BoundsError(A, i))
+    nothing
+end
+# checkbounds in indexing via CartesianIndex
+Base.checkbounds(A::LinearMap, i::Union{CartesianIndex{2}, AbstractArray{CartesianIndex{2}}}) =
+    Base.checkbounds_indices(Bool, axes(A), (i,)) || throw(BoundsError(A, i))
+Base.checkbounds(A::LinearMap, I::AbstractMatrix{Bool}) =
+    axes(A) == axes(I) || throw(BoundsError(A, I))
+
+# main entry point
+function Base.getindex(A::LinearMap, I...)
+    @boundscheck checkbounds(A, I...)
+    _getindex(A, Base.to_indices(A, I)...)
+end
+# quick pass forward
+Base.@propagate_inbounds Base.getindex(A::ScaledMap, I...) = A.λ * A.lmap[I...]
+Base.@propagate_inbounds Base.getindex(A::WrappedMap, I...) = A.lmap[I...]
+
+########################
+# linear indexing
+########################
+_getindex(A::LinearMap, _) = error("linear indexing of LinearMaps is not supported")
+
+########################
+# Cartesian indexing
+########################
+_getindex(A::LinearMap, i::Integer, j::Integer) =
+    error("scalar indexing of LinearMaps is not supported, consider using A[:,j][i] instead")
+_getindex(A::LinearMap, I::Indexer, j::Integer) =
+    error("partial vertical slicing of LinearMaps is not supported, consider using A[:,j][I] instead")
+_getindex(A::LinearMap, ::Base.Slice, j::Integer) = A*unitvec(A, 2, j)
+_getindex(A::LinearMap, i::Integer, J::Indexer) =
+    error("partial horizontal slicing of LinearMaps is not supported, consider using A[i,:][J] instead")
+function _getindex(A::LinearMap, i::Integer, J::Base.Slice)
+    try
+        # requires adjoint action to be defined
+        return vec(unitvec(A, 1, i)'A)
+    catch
+        error("horizontal slicing A[$i,:] requires the adjoint of $(typeof(A)) to be defined")
+    end
+end
+_getindex(A::LinearMap, I::Indexer, J::Indexer) =
+    error("partial two-dimensional slicing of LinearMaps is not supported, consider using A[:,J][I] or A[I,:][J] instead")
+_getindex(A::LinearMap, ::Base.Slice, ::Base.Slice) = convert(AbstractMatrix, A)
+_getindex(A::LinearMap, I::Base.Slice, J::Indexer) = __getindex(A, I, J)
+_getindex(A::LinearMap, I::Indexer, J::Base.Slice) = __getindex(A, I, J)
+function __getindex(A, I, J)
+    dest = zeros(eltype(A), Base.index_shape(I, J))
+    # choose whichever requires less map applications
+    if length(I) <= length(J)
+        try
+            # requires adjoint action to be defined
+            _fillbyrows!(dest, A, I, J)
+        catch
+            error("wide slicing A[I,J] with length(I) <= length(J) requires the adjoint of $(typeof(A)) to be defined")
+        end
+    else
+        _fillbycols!(dest, A, I, J)
+    end
+    return dest
+end
+
+# helpers
+function unitvec(A, dim, i)
+    x = zeros(eltype(A), size(A, dim))
+    @inbounds x[i] = one(eltype(A))
+    return x
+end
+
+function _fillbyrows!(dest, A, I, J)
```

Comment:
should these be commented out since `A[I,J]` is not supported?
the codecov warning is what caught my eye here.

Reply:
The tests were not variable enough. This method is needed in general.